actors/miner/src/lib.rs

```diff
@@ -5033,56 +5023,60 @@ fn batch_activate_deals_and_claim_allocations(
                 sector_expiry: activation_info.sector_expiry,
             })
             .collect();
-        sectors_claims.append(&mut sector_claims);
+        verified_claims.append(&mut sector_claims);
     }
 
-    let claim_res = match sectors_claims.is_empty() {
-        true => Vec::default(),
+    let claim_res = match verified_claims.is_empty() {
+        true => ext::verifreg::ClaimAllocationsReturn {
+            claim_results: BatchReturn::empty(),
+            claims: Vec::default(),
+        },
         false => {
             let claim_raw = extract_send_result(rt.send_simple(
                 &VERIFIED_REGISTRY_ACTOR_ADDR,
                 ext::verifreg::CLAIM_ALLOCATIONS_METHOD,
                 IpldBlock::serialize_cbor(&ext::verifreg::ClaimAllocationsParams {
-                    allocations: sectors_claims,
+                    allocations: verified_claims,
                     all_or_nothing: true,
                 })?,
                 TokenAmount::zero(),
-            ))?;
-
-            let claim_res: ext::verifreg::ClaimAllocationsReturn = deserialize_block(claim_raw)
-                .with_context_code(ExitCode::USR_ILLEGAL_ARGUMENT, || {
-                    "failed to claim allocations during batch activation"
-                })?;
+            ))
+            .context(format!("error claiming allocations on batch {:?}", activation_infos))?;
 
-            claim_res.claim_results
+            let claim_res: ext::verifreg::ClaimAllocationsReturn = deserialize_block(claim_raw)?;
+            claim_res
         }
     };
 
-    // claim res is an iterator of successful claim results parallel to `sector_claims`
-    let mut claim_res = claim_res.into_iter();
+    assert!(
+        claim_res.claim_results.all_ok() || claim_res.claim_results.success_count == 0,
```

Comment:
The all or nothing flag causes the whole ClaimAllocations call to fail if there are any failures, rather than successfully return with success_count = 0. So the second condition here shouldn't happen. Reduce the assertion to just `all_ok()`.